### PR TITLE
Adds tags to discussions (back-end only)

### PR DESCRIPTION
**Pull request by naps62**
The input for tags already existed on the front-end, and on the database, but wasn't being used anywhere.
This sets up the back-end to change that

We haven't really how we want to use tags (or if we want them at all), but this is a mostly generic solution (and was made during an apprenticeship office visit)

Changes:

* Removes the tags field from the database, in favor of https://github.com/mbleigh/acts-as-taggable-on
* Changes the field name from `tags` to `tag_list`, to match the defaults of the gem
* Adds the list of tags to the serializer

### CODE REVIEW

File: app/assets/javascripts/app/discussions/discussion_form.js [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
   function DiscussionFormCtrl(Discussion, $state, $mdToast, $scope, discussion) {
     var ctrl = this;
-
+
```
Comment (joaojusto) [inline app/assets/javascripts/app/discussions/discussion_form.js]: Trailing white space

File: app/assets/javascripts/application.js [hunk 2]
```diff
@@ -27,4 +27,4 @@
 //= require ./app/members/members.module
 //= require_tree ./app/members
 //= require ./app/users/users.module
-//= require_tree ./app/users
+//= require_tree ./app/users
```
Comment (joaojusto) [inline app/assets/javascripts/application.js]: No end line at end of file

**Comment by joaojusto** [comment]
After those small corrections :+1:

File: app/assets/javascripts/app/discussions/discussion_form.js (hunk 1 above)
Comment (naps62) [inline app/assets/javascripts/app/discussions/discussion_form.js]: fixed

File: app/assets/javascripts/application.js (hunk 2 above)
Comment (naps62) [inline app/assets/javascripts/application.js]: fixed

**Pull request merged**